py3status/py3.py

```diff
@@ -679,3 +702,62 @@ def threshold_get_color(self, value, name=None):
         setattr(self._py3status_module, color_name, color)
 
         return color
+
+    def get_url(self, url):
+        """
+        Get the HTTP Response for the HTTP GET request to `url`.
+
+        Raises `Py3.URLError` on Connection Errors.
+        Raises `Py3.HttpClientError` on HTTP 4xx Errors.
+        Raises `Py3.HttpServerError` on HTTP 5xx Errors.
+        """
+        try:
+            response = urlopen(url)
+        except URLError as e:
+            self.log('Unable to access [{}]'.format(url),
+                     level=self.LOG_ERROR)
+            raise self.URLError(e)
+        return_code = response.getcode()
+        status_message = '{url} returned a {http_status} status.'.format(
+            url=url, http_status=return_code)
+        if 400 <= return_code < 500:
+            self.log(status_message, level=self.LOG_ERROR)
+            raise self.HttpClientError(status_message)
+        elif 500 <= return_code < 600:
+            self.log(status_message, level=self.LOG_WARNING)
+            raise self.HttpServerError(status_message)
+        return response
+
+    def get_json_data(self, url):
+        """
+        Deserialize JSON from `url` into a Python object.
+
+        `url` can be str or a HTTP Response object.
+
+        Returns JSON object on success, or None on error.
+        Does not raise any exceptions to caller.
+        """
+        if hasattr(url, 'getcode'):
+            response = url
+        else:
+            try:
+                response = self.get_url(url)
+            except self.Exceptions:
+                # Any Exceptions extended from `Py3.Exceptions`
+                return None
+            except Exception as e:
+                self.log('An uncaught exception was raised [{}]'.format(e.with_traceback()),
+                         level=self.LOG_ERROR)
+                return None
+        reader = codecs.getreader('utf-8')
+        try:
+            data = json.load(reader(response))
+        except (TypeError, ValueError) as e:
+            self.log('Error in json.load for [URL: {}] [{}]'.format(url, e),
+                     level=self.LOG_ERROR)
+            return None
+        except Exception as e:
+            self.log('An uncaught exception was raised [{}]'.format(e.with_traceback()),
+                     level=self.LOG_ERROR)
```

Comment:
We can actually use py3status' `report_exception()` to do this.  But for now this is ok